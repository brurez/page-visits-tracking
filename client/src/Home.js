import React, { Component } from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';
import moment from 'moment';

const data = [
  {
    _id: 'ad767sdsd',
    name: 'Bruno de Rezende',
    email: 'brurez@hotmail.com',
    visits: [1, 2, 4, 5],
    createdAt: '10-10-2017',
  },
  {
    _id: 'ad7676dsd',
    name: 'Babi Reno',
    email: 'babi@hotmail.com',
    visits: [1],
    createdAt: '10-11-2017',
  },
];

class Home extends Component {
  state = { data: [] };

  componentDidMount() {
    axios
      .get('/api/contacts')
      .then(res => {
        //debugger;
        const { data, count } = res.data;
        this.setState({ data });
      })
      .catch(err => console.log(err));
  }

  renderContacts() {
    const data = this.state.data;
    return data.map(contact => {
      const { name, email, visits, createdAt } = contact;
      //debugger;
      return (
        <div className="card" key={contact._id}>
          <div className="content">
            <div className="header">{name}</div>
            <div className="meta">Criado em: {moment(createdAt).format('LLL')}</div>
            <div className="description">
              <strong>Email:</strong> {email} <br />
              <strong>NÃºmero de visitas: </strong> {visits.length}
            </div>
          </div>
          <div className="extra content">
            <Link
              as="button"
              to={`/contact/${contact._id}`}
              className="ui small icon button"
            >
              Ver visitas <i className="arrow right icon" />
            </Link>
          </div>
        </div>
      );
    });
  }

  render() {
    return (
      <div>
        <h2>Lista de Contatos</h2>
        <div className="ui two cards">{this.renderContacts()}</div>
      </div>
    );
  }
}

export default Home;
