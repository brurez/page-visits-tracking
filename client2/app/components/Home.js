import React, {Component} from 'react';

const data = [
  {
    _id: 'ad767sdsd',
    name: 'Bruno de Rezende',
    email: 'brurez@hotmail.com',
    visits: [1, 2, 4, 5],
    createdAt: new Date(),
  },
  {
    _id: 'ad7676dsd',
    name: 'Babi Reno',
    email: 'babi@hotmail.com',
    visits: [1],
    createdAt: new Date(),
  },
];

class Home extends Component {
  renderContacts() {
    return data.map(contact => {
      const { name, email, visits, createdAt } = contact;
      debugger;
      return (
        <div className="card" key={contact._id}>
          <div className="content">
            <div className="header">{name}</div>
            <div className="meta">Criado em: {createdAt}</div>
            <div className="description">
              <strong>Email:</strong> {email}
              <strong>NÃºmero de visitas: </strong> {visits.length}
            </div>
          </div>
        </div>
      );
    });
  }

  render() {
    return (
      <div>
        <h2>Contact List</h2>
        {this.renderContacts()}
      </div>
    );
  }
}

export default Home;
