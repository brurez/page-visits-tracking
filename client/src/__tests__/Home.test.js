import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Home from '../components/Home';

configure({ adapter: new Adapter() });

describe('Home Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  test('if exist', () => {
    expect(wrapper).toBeTruthy();
  });

  test('if have two h2 headers', () => {
     const headers2 = wrapper.find('h2');
     expect(headers2.length).toBe(2);
     expect(headers2.at(0).text()).toContain('Ãšltimas 10 Visitas');
     expect(headers2.at(1).text()).toContain('Lista de Contatos');
  });

  test('match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
