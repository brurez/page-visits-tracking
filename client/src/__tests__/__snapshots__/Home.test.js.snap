// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component match the snapshot 1`] = `
<div>
  <div
    className="ui segment"
  >
    <h2>
      <i
        className="desktop icon"
      />
       Ãšltimas 10 Visitas
    </h2>
    <Visits />
  </div>
  <div
    className="ui segment"
  >
    <h2>
      <i
        className="users icon"
      />
       Lista de Contatos
    </h2>
    <ContactList />
  </div>
</div>
`;
